### ISSUE TITLE
Issue with uploading file using Slack client

### ISSUE THREAD

**Issue by Xorso** [errbotio/errbot#932]
In order to let us help you better, please fill out the following fields as best you can:

### I am...

* [x] Reporting a bug
* [ ] Suggesting a new feature
* [ ] Requesting help with running my bot
* [ ] Requesting help writing plugins
* [ ] Here about something else

### I am running...

* Errbot version: 4.3.4
* OS version: macOS 10.12.1
* Python version: 3.5.0
* Using a virtual environment: yes

### Issue description

Error when posting a file attachment to slack:

Error: Slack API call to files.upload failed: no_file_data

### Steps to reproduce

Submit a BytesIO file to the stream

```python
self.send_stream_request(
                msg.frm,
                BytesIO(r.content),
                name=filename,
            )
```

### Additional info

It looks like the issue may be within the plugin. I wanted to verify before submitting a pull request

Line: 573 in slack.py
```python
            resp = self.api_call('files.upload', data={
                'channels': stream.identifier.channelid,
                'filename': stream.name,
                'file': stream
                })
```

I believe the issue is with `'file': stream` it should be `'file': stream.raw`.

**Comment by jasedit** [comment]
I'm poking around at this to try and help - do you happen to know what the type of `r.content` is when you call `send_stream_request`? If it's a file object, does the code work if you don't wrap it in a `BytesIO` object?

**Comment by andrewthetechie** [comment]
@Xorso we've merged a fix for this issue in #1283 

#closeit
